Runner/Runnable implementation

4ndr3w opened the pull request:
Implements a `Runnable` class that can be scheduled in a `Runner`.

Not 100% on the implementation yet, but this is a starting point.

@ryanpitt thoughts?

4ndr3w committed:
First shot at a Runner/Runnable implementation

4ndr3w committed:
Style fix

4ndr3w committed:
Updated linked list implementation

4ndr3w committed:
Fixed build on GCC

ryanpitt commented on include/motion_control/instance_list_node.h:
```diff
@@ -0,0 +1,65 @@
+
+    InstanceListNode<T>* get_next();
+    InstanceListNode<T>* get_prev();
+    bool in_list();
```
We had a discussion earlier about making some sort of iterator for this. Not necessary to change it now, but let's put it on the backburner for possible future development.

ryanpitt commented on src/runner.cpp:
```diff
@@ -0,0 +1,32 @@
+    double dt = Timer::now() - lastTime;
+
+    InstanceListNode<Runnable*> *node = list.get_root();
+    while ( node != 0 )
```
Same as the other comment, let's user `nullptr` for things like this.

ghost commented:
Let's also rebase onto master after the #7 merge to verify it passes tests.

4ndr3w committed:
Address code review issues on runner

Using std::nullptr for NULL
Linked list add/remove return true/false on success/failure
Runner add/remove returns the success of the operation on the linked list
Style fixes

4ndr3w committed:
Missed some nullptr changes
